src/app/components/EmailNotificationSettingsDialog/components/ErrorMessage.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useMemo } from 'react';
+import { ErrorBadge } from 'app/components/Form/ErrorBadge';
+import { translations } from 'locales/i18n';
+import { useTranslation } from 'react-i18next';
+
+interface ErrorMessageProps {
+  isValidEmail?: boolean;
+  hasUnconfirmedEmail?: boolean;
+}
+
+export const ErrorMessage: React.FC<ErrorMessageProps> = ({
+  isValidEmail,
+  hasUnconfirmedEmail,
+}) => {
+  const { t } = useTranslation();
+  const renderMessage = useMemo(() => {
+    if (!isValidEmail) {
+      return t(translations.emailNotificationsDialog.invalidEmailWarning);
+    }
+    if (hasUnconfirmedEmail) {
+      return t(translations.emailNotificationsDialog.unconfirmedEmailWarning);
+    }
+    return '';
+  }, [t, isValidEmail, hasUnconfirmedEmail]);
```

Comment:
```suggestion
  }, [isValidEmail, hasUnconfirmedEmail]);
```

If you use `import { t } from 'i18next';` then you don't need to use the hook or add to dependency array

Reply:
did you try to do this on D1? because I got an error - `Module '"i18next"' has no exported member 't'.`  Of course i can do it like on D2 but it needs additional changes :wink:

Reply:
yay old code 😄 you can leave it as-is then